Common/src/main/java/com/almostreliable/unified/utils/JsonCompare.java

```diff
@@ -270,6 +268,17 @@ public String getName() {
         }
     }
 
+    public record CompareContext(CompareSettings settings, List<String> compareFields) {
+        public static CompareContext create(CompareSettings settings, RecipeLink curRecipe) {
+            Set<String> compareFields = curRecipe.getActual().keySet();
+            if (!settings.ignoredFields.isEmpty()) {
+                compareFields = new HashSet<>(compareFields);
+                compareFields.removeAll(settings.ignoredFields);
```

Comment:
Checking the ignored fields happens so frequently that it slows down loading.
It is much faster to do it once here and keeping a list of the fields we need to compare.